File: foa/springapp/src/main/java/com/examly/springapp/service/FoodService.java
package com.examly.springapp.service;

import com.examly.springapp.model.Food;
import com.examly.springapp.model.Restaurant;
import com.examly.springapp.repository.FoodRepository;
import com.examly.springapp.repository.RestaurantRepository;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.stream.Collectors;

@Service
public class FoodService {

    @Autowired
    private FoodRepository foodRepo;

    @Autowired
    private RestaurantRepository restaurantRepo;

    /* ================= ADD FOOD ================= */
    public Food addFood(Food food, Long restaurantId) {
        Restaurant restaurant = restaurantRepo.findById(restaurantId)
                .orElseThrow(() -> new RuntimeException("Restaurant not found"));
        food.setRestaurant(restaurant);
        return foodRepo.save(food);
    }

    /* ================= UPDATE FOOD ================= */
    public Food updateFood(Food food) {
        return foodRepo.save(food);
    }

    /* ================= DELETE FOOD ================= */
    public void deleteFood(Long foodId) {
        Food food = foodRepo.findById(foodId).orElseThrow(() -> new RuntimeException("Food not found"));
        foodRepo.delete(food);
    }

    /* ================= GET ALL FOODS ================= */
    public List<Food> getAllFoods() {
        return foodRepo.findAll();
    }

    /* ================= GET FOODS BY RESTAURANT ================= */
    public List<Food> getFoodsByRestaurant(Long restaurantId) {
        return foodRepo.findAll()
                .stream()
                .filter(f -> f.getRestaurant().getId().equals(restaurantId))
                .collect(Collectors.toList());
    }
}

File: foa/reactapp/src/services/foodService.js
import api from "./api";

// export const addFood = (food, restaurantId) =>
//   //api.post(`/api/hotel/foods?restaurantId=${restaurantId}`, food);
//   api.post(`/api/hotel/foods/${restaurantId}`, food);
export const addFood = (food, restaurantId) =>
  api.post(`/api/hotel/foods?restaurantId=${restaurantId}`, food);

export const getFoodsByRestaurant = (restaurantId) =>
  api.get(`/api/foods?restaurantId=${restaurantId}`);

export const updateFood = (food) =>
  api.put(`/api/foods`, food);

export const deleteFood = (foodId) =>
  api.delete(`/api/foods/${foodId}`);

